test/ILLink.RoslynAnalyzer.Tests/DynamicallyAccessedMembersAnalyzerTests.cs

```diff
@@ -19,6 +21,568 @@ static Task VerifyDynamicallyAccessedMembersAnalyzer (string source, params Diag
 				expected: expected);
 		}
 
+		static Task VerifyDynamicallyAccessedMembersCodeFix (
+			string source,
+			string fixedSource,
+			DiagnosticResult[] baselineExpected,
+			DiagnosticResult[] fixedExpected,
+			int? numberOfIterations = null)
+		{
+			var test = new VerifyCS.Test {
+				TestCode = source,
+				FixedCode = fixedSource,
+				ReferenceAssemblies = TestCaseUtils.Net6PreviewAssemblies
+			};
+			test.ExpectedDiagnostics.AddRange (baselineExpected);
+			test.TestState.AnalyzerConfigFiles.Add (
+						("/.editorconfig", SourceText.From (@$"
+is_global = true
+build_property.{MSBuildPropertyOptionNames.EnableTrimAnalyzer} = true")));
+			if (numberOfIterations != null) {
+				test.NumberOfIncrementalIterations = numberOfIterations;
+				test.NumberOfFixAllIterations = numberOfIterations;
+			}
+			test.FixedState.ExpectedDiagnostics.AddRange (fixedExpected);
+			return test.RunAsync ();
+		}
+#region CodeFixTests
+		[Fact]
+		public async Task DiagID_2070 ()
```

Comment:
Nit: the name of the tests is kind of not consistent (DiagID_2070 vs DAM2092 vs MismatchParameter2069). You also need to consider that since these are [Facts] they will appear in the TestExplorer as individual entries, and having a descriptive name does give a reward in developer experience. For example, I would list in the name that these tests are about code fixing stuff. Since now DynamicallyAccessedMembersAnalyzerTests has both, analysis-only tests and code fix tests.

Reply:
I've standardized the names. but do you have any suggestions on how to make them specific? I'm not sure that the diagnostic ID number is very important, should I just include something more like `CodeFix_MismatchParamTargetsParam_NonPublicMethods` if I'm referring to ID 2070 and looking at a case that uses `NonPublicMethods`?

Reply:
My preference here is that is better to give as many clues as possible, even adding the id is fine since they don't tend to change. Something like `CodeFix_IL2070_MismatchParamTargetsParam_NonPublicMethods`